pages/profiles/[profile].js

```diff
@@ -1,21 +1,470 @@
+  // HTML id of the JSON panel
+  panelId: PropTypes.string.isRequired,
+  // True if the JSON panel is open
+  isJsonDetailOpen: PropTypes.bool.isRequired,
+};
+
+/**
+ * Display a single property dependency with a toggle to show the JSON detail.
+ */
+function Dependency({
+  dependencyProperty,
+  dependencies,
+  isJsonDetailOpen,
+  setJsonDetailOpen,
+}) {
+  const dependency = dependencies[dependencyProperty];
+  const dependencyControlId = `dependency-${dependencyProperty}-control`;
+  const dependencyPanelId = `dependency-${dependencyProperty}`;
+
+  return (
+    <div className="my-6 first:mt-0 last:mb-0">
+      <button
+        id={dependencyControlId}
+        onClick={(e) => setJsonDetailOpen(dependencyProperty, e.altKey)}
+        className="flex items-center pr-2 text-sm font-bold text-schema-prop"
+        aria-label={`${dependencyProperty} dependency`}
+        aria-expanded={isJsonDetailOpen}
+        aria-controls={dependencyPanelId}
+      >
+        {isJsonDetailOpen ? (
+          <MinusIcon className="mr-1 h-4 w-4" />
+        ) : (
+          <PlusIcon className="mr-1 h-4 w-4" />
+        )}
+        <div className="break-all text-left">{dependencyProperty}</div>
+      </button>
+      <div className="ml-5 text-sm text-schema-prop-description">
+        {dependency.comment}
+      </div>
+      <SchemaJsonPanel
+        property={dependency}
+        panelId={dependencyPanelId}
+        isJsonDetailOpen={isJsonDetailOpen}
+      />
+    </div>
+  );
+}
+
+Dependency.propTypes = {
+  // Property that has dependencies
+  dependencyProperty: PropTypes.string.isRequired,
+  // Single property dependency to display
+  dependencies: PropTypes.object.isRequired,
+  // True if the JSON panel is open
+  isJsonDetailOpen: PropTypes.bool.isRequired,
+  // Callback to set the JSON panel open state
+  setJsonDetailOpen: PropTypes.func.isRequired,
+};
+
+/**
+ * Section of the display that shows the property dependencies of a schema.
+ */
+function Dependencies({ dependencies = null }) {
+  // Contains the names of all properties with their JSON panels open
+  const [openJsonIds, setOpenJsonIds] = useState([]);
+
+  // Callback to set the JSON panel open state for property dependencies.
+  function handleJsonPanelOpen(propertyId, isOptionKey) {
+    if (openJsonIds.includes(propertyId)) {
+      // The clicked property is open. Remove it from the list of open panels, or close all panels
+      // if the user holds down the option key.
+      if (isOptionKey) {
+        setOpenJsonIds([]);
+      } else {
+        setOpenJsonIds(openJsonIds.filter((id) => id !== propertyId));
+      }
+    } else {
+      // The clicked property is closed. Add it to the list of open panels, or open all panels if
+      // the user holds down the option key.
+      if (isOptionKey) {
+        setOpenJsonIds(Object.keys(dependencies));
+      } else {
+        setOpenJsonIds([...openJsonIds, propertyId]);
+      }
+    }
+  }
+
+  if (dependencies) {
```

Comment:
somehow there is empty dependentSchemas in some of the schemas. So we should also check if the dependentSchemas is empty. Check the biomarker has empty dependentSchemas: https://igvf-ui-igvf-1310-formatted-schemas.demo.igvf.org/profiles/biomarker/

Reply:
Did not even realize `dependentSchemas` wasn’t consistently defined.